### PR TITLE
PaddleRec release 1.6 api update

### PR DESCRIPTION
**Pull request by wilhelmzh**
Update API under PaddleRec/ctr, including dnn, dcn, deepfm and xdeepfm.
Change README to remind user to use Paddle >= 1.6

### CODE REVIEW

File: PaddleRec/ctr/dcn/network.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def _l2_loss(self, w):
 
     def _create_embedding_input(self, data_dict):
         # sparse embedding
-        sparse_emb_dict = OrderedDict((name, fluid.layers.embedding(
+        sparse_emb_dict = OrderedDict((name, fluid.embedding(
```
Comment (Aurelius84) [inline PaddleRec/ctr/dcn/network.py:123]: fluid.embeding将在输入的shape后直接append一维embed_size。如[None, 1] -> [None, 1, embed_size]
请确认是否符合预期？

File: PaddleRec/ctr/deepfm/README.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ This model implementation reproduces the result of the paper "DeepFM: A Factoriz
 ```
 
 ## Environment
-- PaddlePaddle 1.6
+- **PaddlePaddle version >= 1.6, or develop version**
```
Comment (Aurelius84) [inline PaddleRec/ctr/deepfm/README.md:19]: **Now all models in PaddleRec require PaddlePaddle version 1.6 or higher, or suitable develop version.**

File: PaddleRec/ctr/deepfm/network_conf.py [hunk 3]
```diff
@@ -25,7 +25,7 @@ def ctr_deepfm_model(embedding_size,
 
     # -------------------- first order term  --------------------
 
-    first_weights_re = fluid.layers.embedding(
```
Comment (Aurelius84) [inline PaddleRec/ctr/deepfm/network_conf.py:28]: feat_idx最后一维为1，旧接口会在此维度上直接替换为embed_size。新的接口fluid.embedding是直接append embed_size作为输出shape，请确认是否和更改前一致？

File: PaddleRec/ctr/dnn/README.md [hunk 4]
```diff
@@ -20,6 +20,7 @@ factorization machines, please refer to the paper [factorization
 machines](https://www.csie.ntu.edu.tw/~b97053/paper/Rendle2010FM.pdf)
 
 ## Environment
+**PaddlePaddle version should >= 1.6, or using develop version.**
```
Comment (Aurelius84) [inline PaddleRec/ctr/dnn/README.md]: **Now all models in PaddleRec require PaddlePaddle version 1.6 or higher, or suitable develop version.**

File: PaddleRec/ctr/dcn/network.py (hunk 1 above)
Comment (wilhelmzh) [reply PaddleRec/ctr/dcn/network.py:123]: 后面做了flatten，所以这里没加squeeze

File: PaddleRec/ctr/xdeepfm/network_conf.py [hunk 5]
```diff
@@ -14,18 +14,18 @@ def ctr_xdeepfm_model(embedding_size,
...
+        name='feat_value', shape=[None, num_field], dtype='float32')
+    label = fluid.data(
+        name='label', shape=[None, 1], dtype='float32')  # None * 1
     feat_idx = fluid.layers.reshape(raw_feat_idx,
                                     [-1, 1])  # (None * num_field) * 1
     feat_value = fluid.layers.reshape(
         raw_feat_value, [-1, num_field, 1])  # None * num_field * 1
 
-    feat_embeddings = fluid.layers.embedding(
+    feat_embeddings = fluid.embedding(
```
Comment (Aurelius84) [inline PaddleRec/ctr/xdeepfm/network_conf.py:28]: 请确认是否需要对embedding后的结果squeeze下

File: PaddleRec/ctr/dcn/README.md [hunk 6]
```diff
@@ -23,7 +24,7 @@
 DCN模型介绍可以参阅论文[Deep & Cross Network for Ad Click Predictions](https://arxiv.org/abs/1708.05123)
 
 ## 环境
-- PaddlePaddle 1.6
+- 目前模型库下模型均要求使用PaddlePaddle 1.6及以上版本或适当的develop版本
```
Comment (Aurelius84) [inline PaddleRec/ctr/dcn/README.md:28]: 后面的都对此加粗处理了，这个也统一加粗吧

File: PaddleRec/ctr/deepfm/network_conf.py (hunk 3 above)
Comment (wilhelmzh) [reply PaddleRec/ctr/deepfm/network_conf.py:28]: 这里后面有reshape，所以没有squeeze

File: PaddleRec/ctr/xdeepfm/network_conf.py (hunk 5 above)
Comment (wilhelmzh) [reply PaddleRec/ctr/xdeepfm/network_conf.py:28]: 这里也是后面原来就有reshape，所以没用squeeze

File: PaddleRec/ctr/deepfm/README.md (hunk 2 above)
Comment (wilhelmzh) [reply PaddleRec/ctr/deepfm/README.md:19]: solved

File: PaddleRec/ctr/dcn/README.md (hunk 6 above)
Comment (wilhelmzh) [reply PaddleRec/ctr/dcn/README.md:28]: solved

**Review by Aurelius84 — APPROVED** [review]
LGTM

**Pull request merged**